hibernation-setup-tool.c

```diff
@@ -824,6 +824,28 @@ static void perform_fs_specific_checks(const char *path)
     }
 }
 
+bool check_if_system_version_is_higher(const char *version) {
+    struct utsname my_uname;
+    if(uname(&my_uname) == -1)
+    {
+        log_info("uname call failed. Proceeding with fallocate.");
+        return true;
+    }
+    unsigned sys_major_version = 0, sys_minor_version = 0;
+    unsigned req_major_version = 0, req_minor_version = 0;
+    sscanf(my_uname.release, "%u.%u", &sys_major_version, &sys_minor_version);
+    sscanf(version, "%u.%u", &req_major_version, &req_minor_version);
+    if (sys_major_version > req_major_version)
+        return true;
+    if (sys_major_version < req_major_version)
+        return false;
+    if (sys_minor_version > req_minor_version)
+        return true;
+    if (sys_minor_version < req_minor_version)
+        return false;
+    return true;
```

Comment:
This should compare both major and minor in a single go and should take into consideration all the corner cases:

```suggestion
   uint64_t sys_version = sys_major_version * 10000 + sys_minor_version;
   uint64_t req_version = req_major_version * 10000 + req_minor_version;
   return sys_version >= req_version;
```